build/commands/lib/envConfig.test.js

```diff
@@ -0,0 +1,555 @@
+      const configDir = '/path/to/config'
+      const packageJsonPath = path.join(configDir, 'package.json')
+      const envPath = path.join(configDir, '.env')
+
+      const packageData = {
+        version: '1.0.0',
+        config: {},
+      }
+
+      mockFiles[packageJsonPath] = JSON.stringify(packageData)
+
+      const envConfig = new EnvConfig(configDir)
+
+      expect(envConfig).toBeDefined()
+      expect(fs.writeFileSync).toHaveBeenCalledWith(
+        envPath,
+        expect.stringContaining('placeholder .env config file'),
+      )
+    })
+  })
+
+  describe('getPackageConfig', () => {
+    let envConfig
+
+    beforeEach(() => {
+      const configDir = '/path/to/config'
+      const packageJsonPath = path.join(configDir, 'package.json')
+      const envPath = path.join(configDir, '.env')
+
+      const packageData = {
+        version: '1.2.3',
+        config: {
+          projects: {
+            chrome: {
+              version: '120.0.0',
+              tag: 'v120',
+            },
+          },
+          target_arch: 'x64',
+        },
+      }
+
+      mockFiles[envPath] = ''
+      mockFiles[packageJsonPath] = JSON.stringify(packageData)
+
+      envConfig = new EnvConfig(configDir)
+    })
+
+    it('should retrieve nested config values', () => {
+      const result = envConfig.getPackageConfig([
+        'projects',
+        'chrome',
+        'version',
+      ])
+      expect(result).toBe('120.0.0')
+    })
+
+    it('should retrieve top-level config values', () => {
+      const result = envConfig.getPackageConfig(['version'])
+      expect(result).toBe('1.2.3')
+    })
+
+    it('should return undefined for non-existent keys', () => {
+      const result = envConfig.getPackageConfig(['nonexistent', 'key'])
+      expect(result).toBeUndefined()
+    })
+
+    it('should return undefined when path traverses non-object', () => {
+      const result = envConfig.getPackageConfig(['version', 'nested'])
+      expect(result).toBeUndefined()
+    })
+  })
+
+  describe('getEnvConfig', () => {
+    let envConfig
+
+    beforeEach(() => {
+      const configDir = '/path/to/config'
+      const packageJsonPath = path.join(configDir, 'package.json')
+      const envPath = path.join(configDir, '.env')
+
+      const packageData = {
+        version: '1.2.3',
+        config: {
+          fallback_value: 'from_package',
+        },
+      }
+
+      mockFiles[packageJsonPath] = JSON.stringify(packageData)
+      mockFiles[envPath] = [
+        'STRING_VALUE=hello',
+        'NUMBER_VALUE=42',
+        'BOOL_VALUE=true',
+        'ARRAY_VALUE=["a","b","c"]',
+        'OBJECT_VALUE={"key":"value"}',
+      ].join('\n')
+
+      envConfig = new EnvConfig(configDir)
+    })
+
+    it('should retrieve string values from .env', () => {
+      const result = envConfig.getEnvConfig(['STRING', 'VALUE'])
+      expect(result).toBe('hello')
+    })
+
+    it('should parse JSON values when default is not provided', () => {
```

Comment:
it make sense to add a case for a number with the default value, i.e. `envConfig.getEnvConfig(['NUMBER', 'VALUE'], 0)`